### ISSUE TITLE
Return initial game state when creating a game with 1 human player

### ISSUE THREAD

**Issue by jakejgordon** [JoyfulBits/fungus_toast#14]
As a player who just created a game with 1 human player (and hence doesn't need to wait for any other human players to join the game)
I can get the initial game state back immediately
So I know how to render the players' cells on the initial grid

Technical notes:
If you call POST /games and you pass numberOfHumanPlayers = 1, then all the criteria is met to start the game. The 201 response can also include the gameState and latestStateChange indicating where each player's start cell should be placed.

gameState JSON example:
```json
{
  "cells": [
    {
      "index": 30,
      "cellType": 0,
      "playerId": 10
    },
    {
      "index": 31,
      "playerId": 10,
      "cellType": 1
    },
    {
      "index": 32,
      "cellType": 2,
      "playerId": 20,
      "previousplayerId": 10
    }
  ]
}
```
stateChange json example:
```json
{
   "roundNumber":12,
   "stateChanges":[
      {
         "cycle":0,
         "stateChange":{
            "mutationPointsAwarded":[
               {
                  "playerId":1,
                  "mutationPoints":1
               },
               {
                  "playerId":2,
                  "mutationPoints":3
               }
            ],
            "changedCells":[
               {
                  "index":30,
                  "newCellType":0,
                  "playerId":10
               },
               {
                  "index":31,
                  "newCellType":1,
                  "playerId":10,
                  "previousCellType":0
               },
               {
                  "index":32,
                  "newCellType":1,
                  "playerId":20,
                  "previousCellType":0,
                  "previousPlayerId":10
               }
            ]
         }
      },
      {
         "cycle":1,
         "stateChange":{
            "changedCells":[

            ]
         }
      }
   ]
}
```

**Comment by daveshah** [comment]
@jakejgordon I was giving our convo earlier some thought. What would you think about returning `cycles` back in the game state? Each cycle would contain the full grid. The first grid in cycles *could* always be the most current state before the cycles kick off (so if there were 5 cycles, `cycles` would contain 6 grid objects - one for the initial game state, 5 for the changes within each cycle. Something like:
```json
{
   "gameState":{
      "cycles":[
         {
            "grid":[
               [
                  {
                     "cell_type":null,
                     "player_id":null
                  },
                  {
                     "cell_type":null,
                     "player_id":null
                  }
               ],
               [
                  {
                     "cell_type":null,
                     "player_id":null
                  },
                  {
                     "cell_type":null,
                     "player_id":null
                  }
               ]
            ]
         },
         {
            "grid":[
               [
                  {
                     "cell_type":1,
                     "player_id":100
                  },
                  {
                     "cell_type":1,
                     "player_id":100
                  }
               ],
               [
                  {
                     "cell_type":1,
                     "player_id":201
                  },
                  {
                     "cell_type":1,
                     "player_id":201
                  }
               ]
            ]
         }
      ]
   }
}
```

**Comment by jakejgordon** [comment]
I think it could work -- but it still puts more of the onus on the client to identify what actually changed about the state. If you have a 2500 element array and a single cell changes, you've got to iterate over both of them comparing, and then ultimately doing one animation for that one cell. If you got back the initial state and a series of deltas it I think it would be simpler for the client (and lighter weight -- as much as we don't care about that yet :) ).

**Comment by daveshah** [comment]
🤔I see what you mean, but I'd like to try and avoid book-keeping if we can with deltas. We should keep iterating on this one, but as we do, maybe actually make some client calls on those iterations to see what this feels like.

**Comment by jakejgordon** [comment]
I think this issue can be closed out once [the ex machina branch](https://github.com/JoyfulBits/fungus_toast/tree/ex-machina-setup) is wrapped up.

**Comment by jakejgordon** [comment]
Finally closed by #24
